# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gzip/no_basename.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


That's not a nice but working patch to let gzip compile with glibc.

 - Clifford

diff -ruN ../gzip-1.2.4/gzip.h ./gzip.h
--- ./../gzip-1.2.4/gzip.h	Fri Aug 13 14:35:33 1993
+++ ./gzip.h	Thu May 28 00:01:01 1998
@@ -301,7 +301,7 @@
 extern void flush_window  OF((void));
 extern void write_buf     OF((int fd, voidp buf, unsigned cnt));
 extern char *strlwr       OF((char *s));
-extern char *basename     OF((char *fname));
+extern char *Xbasename    OF((char *fname));
 extern void make_simple_name OF((char *name));
 extern char *add_envopt   OF((int *argcp, char ***argvp, char *env));
 extern void error         OF((char *m));
diff -ruN ../gzip-1.2.4/util.c ./util.c
--- ./../gzip-1.2.4/util.c	Fri Aug 13 14:35:40 1993
+++ ./util.c	Thu May 28 00:01:40 1998
@@ -178,7 +178,7 @@
  * any version suffix). For systems with file names that are not
  * case sensitive, force the base name to lower case.
  */
-char *basename(fname)
+char *Xbasename(fname)
     char *fname;
 {
     char *p;
